### PR TITLE
Add FAQ entry "Which programming language is fastest?"

### PR DESCRIPTION
**Pull request by tetrapod00**
Compares the performance of C#, GDScript, and C++.

Extracts the programming language comparison from https://github.com/godotengine/godot-docs/pull/7538 and moves it to a dedicated FAQ entry. Now feel free to argue about the characterization of your favorite language 😄 .

CC @Calinou, this is largely based on your revised description from that PR.

(At some point I might like to add a dedicated "Which language should I use?" entry to the FAQ, but it's not clean to add right now because the current FAQ is designed to sell you on GDScript. The current "Which programming languages are supported in Godot?" and "What is GDScript and why should I use it?" combined cover this topic already.)

### CODE REVIEW

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
As noted in https://github.com/godotengine/godot-docs/pull/10286, I'm in favor of this PR and removing that comparison from the C# docs.

File: about/faq.rst
Comment (hsvfan-jan) [inline about/faq.rst:197]: ~~Your~~ You

**Comment by dsnopek** [comment]
Is it worth noting that C++ in modules can be faster than C++ in GDExtension (also because of marshalling, like pointed out for C#)? Or is that too much information?

**Review by skyace65 — APPROVED** [review]
Looks good to me, it just needs to be rebased since #10286 was merged.

**Comment by tetrapod00** [comment]
Before merging, I wanted to give Calinou (and maintainers for GDExtension and C#) a bit more of a chance to look over if they find time, since it's based on their previous work. If that doesn't happen, we can always make improvements later

File: about/faq.rst
Comment (AThousandShips) [inline about/faq.rst:182]:
```suggestion
In general, the performance of C# and GDScript are about the same order of magnitude,
```

File: about/faq.rst
Comment (AThousandShips) [inline about/faq.rst:195]:
```suggestion
GDScript, and is slower to develop with.
```
Otherwise it seems like it is a future feature IMO

**Review by AThousandShips — APPROVED** [review]

**Comment by skyace65** [comment]
Thanks!

**Comment by mhilbrunner** [comment]
Cherrypicked to 4.3 in https://github.com/godotengine/godot-docs/pull/10347.

**Pull request merged**
